src/logic/community-voice/community-voice.ts

```diff
@@ -0,0 +1,213 @@
+    // We don't need to call the comms-gatekeeper here
+
+    // Analytics event
+    analytics.fireEvent(AnalyticsEvent.END_CALL, {
+      call_id: voiceChatId
+    })
+  }
+
+  async function joinCommunityVoiceChat(communityId: string, userAddress: string): Promise<{ connectionUrl: string }> {
+    logger.info(`User ${userAddress} joining community voice chat for community ${communityId}`)
+
+    // Get the active voice chat for the community
+    const voiceChatStatus = await commsGatekeeper.getCommunityVoiceChatStatus(communityId)
+    if (!voiceChatStatus?.isActive) {
+      throw new CommunityVoiceChatNotFoundError(communityId)
+    }
+
+    // Check if user is member of the community (or if community is public)
+    const community = await communitiesDb.getCommunity(communityId, userAddress)
+    if (!community) {
+      throw new UserNotCommunityMemberError(userAddress, communityId)
+    }
+
+    // Determine user role (listener by default for members, speaker for moderators/owners)
+    const userRole = await communitiesDb.getCommunityMemberRole(communityId, userAddress)
+    const voiceChatRole =
+      userRole === CommunityRole.Owner || userRole === CommunityRole.Moderator
+        ? CommunityVoiceChatRole.SPEAKER
+        : CommunityVoiceChatRole.LISTENER
+
+    // Get credentials from comms-gatekeeper
+    const credentials = await commsGatekeeper.getCommunityVoiceChatCredentials(communityId, userAddress)
+
+    return credentials
+  }
+
+  async function leaveCommunityVoiceChat(voiceChatId: string, userAddress: string): Promise<void> {
+    logger.info(`User ${userAddress} leaving community voice chat ${voiceChatId}`)
+
+    // In our new architecture, voiceChatId is the same as communityId
+    const communityId = voiceChatId
+
+    // Get current voice chat status
+    const voiceChatStatus = await commsGatekeeper.getCommunityVoiceChatStatus(communityId)
+    if (!voiceChatStatus?.isActive) {
+      throw new CommunityVoiceChatNotFoundError(voiceChatId)
+    }
+
+    // Note: We delegate the participant check to the comms-gatekeeper
+    // The gatekeeper will handle whether the user is actually in the voice chat
+
+    // Note: No longer publishing leave events - only 'started' events are published
+  }
+
+  async function getCommunityVoiceChat(communityId: string): Promise<CommunityVoiceChat | null> {
+    const status = await commsGatekeeper.getCommunityVoiceChatStatus(communityId)
+    if (!status?.isActive) {
+      return null
+    }
+
+    // Convert gatekeeper status to our CommunityVoiceChat format
+    return {
+      id: communityId,
+      community_id: communityId,
+      created_by: '', // We don't have this info from gatekeeper, but it's not critical
+      created_at: new Date(), // Same here
+      ended_at: undefined,
+      status: CommunityVoiceChatStatus.ACTIVE
+    }
+  }
+
+  async function getCommunityVoiceChatParticipants(voiceChatId: string): Promise<CommunityVoiceChatParticipant[]> {
```

Comment:
If the user joins the voice chat, the should be able to get the participants from the room. Taking that into consideration, shall we keep this method?